cromshell

```diff
@@ -1124,16 +1124,34 @@ function list()
     tmpFile=$( makeTemp )
     cp ${CROMWELL_SUBMISSIONS_FILE} ${tmpFile}
 
-    while read line ; do 
+    UNREACHABLE_SERVER=""
+    REACHABLE_SERVER=""
+    while read line ; do
       id=$( echo "$line" | awk '{print $3}' )
       # Only update if the ID field is, in fact, a valid ID:
       if [[ $(matchWorkflowId ${id}) -eq 0 ]] ; then
         srv=$( echo "$line" | awk '{print $2}' )
-        runSt=$( echo "$line" | awk '{print $5}' )
-
-        # get the status of the run if it has not completed:
-        if [[ "${runSt}" != "Failed" ]] && [[ "${runSt}" != "Aborted" ]] && [[ "${runSt}" != "Succeeded" ]] ; then
-          status ${id} ${srv} &> /dev/null
+        if echo $UNREACHABLE_SERVER | grep -q $srv; then
+          continue # we already know that server is unreachable
+        elif echo $REACHABLE_SERVER | grep -q $srv; then
+          # we know that server should be reachable, practically speaking during this update operation
+          runSt=$( echo "$line" | awk '{print $5}' )
+
+          # get the status of the run if it has not completed:
+          if [[ "${runSt}" != "Failed" ]] && [[ "${runSt}" != "Aborted" ]] && [[ "${runSt}" != "Succeeded" ]] ; then
+            status ${id} ${srv} &> /dev/null
+          fi
+        elif [[ $(curl -m 1 -s ${srv}) ]]; then
+          REACHABLE_SERVER+=" $srv"
```

Comment:
We try to always use {} around variables ( with the sometimes exception of $1 $2 etc). We're not totally consistent about it but it's good general practice.
```suggestion
          REACHABLE_SERVER+=" ${srv}"
```

Reply:
I believe the new commit does my part.